### PR TITLE
[Improvement-14393][Resource Center] Remove deprecated resource description in api and UI page

### LINKED ISSUE #14393: [Improvement][Resource Center] Remove deprecated resource description in api and UI page
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

In new resource center, description of resource is deprecated.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Radeity**
## Purpose of the pull request
- This PR closes https://github.com/apache/dolphinscheduler/issues/14393

## Verify this pull request

- Update relative UT.

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ResourcesController.java [hunk 1]
```diff
@@ -124,11 +122,10 @@ public class ResourcesController extends BaseController {
     public Result<Object> createDirectory(@Parameter(hidden = true) @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
                                           @RequestParam(value = "type") ResourceType type,
                                           @RequestParam(value = "name") String alias,
-                                          @RequestParam(value = "description", required = false) String description,
                                           @RequestParam(value = "pid") int pid,
                                           @RequestParam(value = "currentDir") String currentDir) {
         // todo verify the directory name
-        return resourceService.createDirectory(loginUser, alias, description, type, pid, currentDir);
```
Comment (zhongjiajie) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ResourcesController.java:131]: could you add some desc to `incompatible.md` due to we change the public interface?

**Review by zhongjiajie — DISMISSED** [review]

**Comment by zhongjiajie** [comment]
restart the failed E2E test

**Comment by zhongjiajie** [comment]
file e2e test case failed, could you make sure it failed by random or code change?

**Comment by Radeity** [comment]
> file e2e test case failed, could you make sure it failed by random or code change?

I've modified e2e test, wait for CI re-run :D

**Review by zhongjiajie — APPROVED** [review]

**Comment by zhongjiajie** [comment]
well done 🎉

**Pull request merged**